app/services/rate_checker.rb

```diff
@@ -1,7 +1,21 @@
 class RateChecker
-  def user_has_already_rated?(submission_id, user_id)
-    submission = Submission.find(submission_id)
+  def initialize(submission_id, user_id)
+    @submission_id = submission_id
+    @user_id = user_id
+  end
 
+  def user_has_already_rated?
+    submission = Submission.find(submission_id)
     submission.rates.any? { |rate| rate.user_id == user_id }
   end
+
+  def current_user_rate_value
+    rate = Rate.find_by(submission_id: submission_id, user_id: user_id)
+    return 0 unless rate
```

Comment:
This is only for marking stars on the starability picker, we don't display this value anywhere. And now since we mark the current user's rate on that picker, it makes sense to not mark any of them if he hasn't rated yet.